modules/payloads/singles/linux/aarch64/meterpreter_reverse_http.rb

```diff
@@ -38,6 +39,11 @@ def generate(_opts = {})
       scheme: 'http',
       stageless: true
     }.merge(mettle_logging_config)
-    MetasploitPayloads::Mettle.new('aarch64-linux-musl', generate_config(opts)).to_binary :exec
+    payload = MetasploitPayloads::Mettle.new('aarch64-linux-musl', generate_config(opts)).to_binary :exec
+    ds = opts[:datastore] || datastore
+    unless ds['MeterpreterLegacyElf']
```

Comment:
If we have access to the datastore, we should be able to tell if the options support the mettle classic payloads.  If no options need the new version, we should use the OG version.
This may be a little harder with venom, but it should be similar.